File: src/main/java/org/bp/lab/simpleblog/service/CommentService.java
package org.bp.lab.simpleblog.service;

import org.bp.lab.simpleblog.domain.Comment;
import org.springframework.data.domain.Page;

import java.util.List;

public interface CommentService {
	
	Page<Comment> findAll(int page, int size);
	Page<Comment> findAllWherePostId(Long id, int page, int size);
	List<Comment> getAllPostComments(Long postId);
	Comment findOne(Long id);
	Comment save(Long postId, Comment comment);
	void delete(Long id);

}

File: src/main/webapp/static/app/js/comment/commentService.js
var blog = angular.module('commentServices', []);

blog.service('commentService', function($http){
	
	this.url = 'api/comments';
	
	this.getPage = function(page, size, postId){
		return $http.get(this.url, { params: {'page': page, 'size': size, 'postId': postId }});
	};
	
	this.findAllByPostId = function(postId){
		return $http.get(this.url, { params: { 'postId': postId }});
	};
	
	this.findAll = function(page, size){
		return $http.get(this.url, { params: { 'page': page, 'size': size }});
	};
	
	this.save = function(comment, postId){
		if(comment.id){
			return $http.put(this.url + '/' + comment.id, comment, { params: { 'postId': postId }});
		}else{
			return $http.post(this.url, comment, { params: { 'postId': postId }});
		}
	};
	
	this.updateLike = function(id, choice){
		return $http.put(this.url + '/like/' + id, {}, { params: { 'choice': choice }});
	};
	
	this.delete = function(id){
		return $http.delete('this.url' + '/' + id);
	};
	
});